samples/secure-sql-chatbot/README.md

```diff
@@ -0,0 +1,55 @@
+# Secure chatbot using advanced RAG and a SQL database
+
+![Movie Muse](moviemuse.png)
+
+This example demonstrates how to create a secure chatbot with RAG using
+`quarkus-langchain4j`. This chatbot internally uses LLM-generated SQL
+queries to retrieve the relevant information from a PostgreSQL database.
+
+## Running the example
+
+Users must authenticate with Google before they can start working with the Movie Muse and register the OpenAI API key.
+
+### Google Authentication
+
+In order to register an application with Google, follow steps listed in the [Quarkus Google](https://quarkus.io/guides/security-openid-connect-providers#google) section.
+Name your Google application as `Quarkus LangChain4j AI`, and make sure an allowed callback URL is set to `https://localhost:8443/login`.
+Google will generate a client id and secret, use them to set `quarkus.oidc.client-id` and `quarkus.oidc.credentials.secret` properties.
+
+### OpenAI API key
+
+You must provide your OpenAI API key:
+
+```
+export QUARKUS_LANGCHAIN4J_OPENAI_API_KEY=<your-openai-api-key>
+```
+
+Then, simply run the project in Dev mode:
+
+```
+mvn quarkus:dev
+```
+
+## Using the example
+
+Open your browser and navigate to `https://localhost:8080`, accept the demo certificate.
+Now, choose a `Login with Google` option.
+Once logged in, click the orange robot in the bottom right corner to open the chat window.
+
+The chatbot is available at the secure `wss:` scheme.
+
+It uses a SQL database with information about movies with their
+basic metadata (the database is populated with data from
+`src/main/resources/data/movies.csv` at startup). When you ask a question, an
+LLM is used to generate SQL queries necessary for answering your question.
+Check the application's log, the SQL queries and the retrieved data will be
+printed there.
+
+The chatbot will refer to you by your name during the conversation.
```

Comment:
The app just prepends the name before the bot's messages, which is not very elegant... I'm thinking how we could make this better. Maybe we could pass the name of the user as part of the system message?! And instead of the hardcoded `"Hello, " + idToken.getName() + ", I'm MovieMuse, how can I help you?"` at start, we could tell the LLM to greet the user...
I don't know - but it would probably make it look better.

Reply:
Hi @jmartisk 

>The app just prepends the name before the bot's messages, which is not very elegant... 

Yeah, I just wanted to confirm it was visible, without thinking how to make it look nicer

>Maybe we could pass the name of the user as part of the system message?! 

Sure, but I'm not sure how we can capture it in a system message ? 

>And instead of the hardcoded "Hello, " + idToken.getName() + ", I'm MovieMuse, how can I help you?" at start, we could tell the LLM to greet the user...

It would be great if we can get it do it, I'm just not sure how to teach LLM that it can get a user name from `idToken.getName()`, what do you think ?

Reply:
Hi @jmartisk , I guess it would not be much of an AI then, which would take a user name as a parameter and echo it back :-).
May be we get that user name used in a more interesting way.... The purpose of the demo is to show the authenticated user identity is visible to AI content retriever,  etc...

Reply:
@jmartisk  We already demo it in the secure fraud detection demo, the demo is kind of similar but the difference is that WebSockets are involved... But yeah, something more interesting with the user name should be shown, somehow it should impact the RAG process output, we can do something simple, just to highlight the point, if you have some ideas, let me know please

Reply:
Yeah I don't have any better idea how to utilize the username other than for greeting the user :/

Reply:
Unless we go really deep and make it into some kind of personalized adviser that knows some specific data about the user, which is out of scope here.

Reply:
Sure @jmartisk, I'll see at least if I can get the user identity impacting the RAG output, thanks